src/og/entity/GeoObjectHandler.js

```diff
@@ -0,0 +1,1105 @@
+        this._sizeBuffer = [];
+        this._vertexBuffer = [];
+        this._positionHighBuffer = [];
+        this._positionLowBuffer = [];
+        this._directionBuffer = [];
+        this._rgbaBuffer = [];
+        this._normalsBuffer = [];
+        this._indicesBuffer = [];
+        this._pickingColorBuffer = [];
+
+        this.__staticId = GeoObjectHandler._staticCounter++;
+        this._buffersUpdateCallbacks = [];
+        this._buffersUpdateCallbacks[PICKINGCOLOR_BUFFER] = this.createPickingColorBuffer;
+        this._buffersUpdateCallbacks[POSITION_BUFFER] = this.createPositionBuffer;
+        this._buffersUpdateCallbacks[DIRECTION_BUFFER] = this.createDirectionBuffer;
+        this._buffersUpdateCallbacks[NORMALS_BUFFER] = this.createNormalsBuffer;
+        this._buffersUpdateCallbacks[RGBA_BUFFER] = this.createRgbaBuffer;
+        this._buffersUpdateCallbacks[INDECIES_BUFFER] = this.createIndicesBuffer;
+        this._buffersUpdateCallbacks[VERTEX_BUFFER] = this.createVertexBuffer;
+        this._buffersUpdateCallbacks[SIZE_BUFFER] = this.createSizeBuffer;
+        this._buffersUpdateCallbacks[PITCH_ROLL_BUFFER] = this.createPitchRollBuffer;
+
+        this._changedBuffers = new Array(this._buffersUpdateCallbacks.length);
+
+        this._instancedTags = new Map();
+        this._instancedTags.set("none", {
+            index: 0
+        });
+    }
+
+    getObjectByIndex(index) {
+        return this._geoObjects[index];
+    }
+
+    //Create buffers
+    createVertexBuffer() {
+        var h = this._renderer.handler;
+        for (let i = 0; i < this._vertexArr.length; i++) {
+            this._vertexBuffer && h.gl.deleteBuffer(this._vertexBuffer[i]);
+            this._vertexArr[i] = makeArrayTyped(this._vertexArr[i]);
+            this._vertexBuffer[i] = h.createArrayBuffer(
+                this._vertexArr[i],
+                3,
+                this._vertexArr[i].length / 3
+            );
+        }
+    }
+
+    createPitchRollBuffer() {
+        var h = this._renderer.handler;
+        for (let i = 0; i < this._pitchRollArr.length; i++) {
+            this._pitchRollBuffer && h.gl.deleteBuffer(this._pitchRollBuffer[i]);
+            this._pitchRollArr[i] = makeArrayTyped(this._pitchRollArr[i]);
+            this._pitchRollBuffer[i] = h.createArrayBuffer(
+                this._pitchRollArr[i],
+                2,
+                this._pitchRollArr[i].length / 2
+            );
+        }
+    }
+
+    createSizeBuffer() {
+        var h = this._renderer.handler;
+        for (let i = 0; i < this._sizeArr.length; i++) {
+            this._sizeBuffer && h.gl.deleteBuffer(this._sizeBuffer[i]);
+            this._sizeArr[i] = makeArrayTyped(this._sizeArr[i]);
+            this._sizeBuffer[i] = h.createArrayBuffer(this._sizeArr[i], 1, this._sizeArr[i].length);
+        }
+    }
+
+    createPositionBuffer() {
+        let h = this._renderer.handler;
+        for (let i = 0; i < this._positionHighArr.length; i++) {
```

Comment:
Looks like we could combine the loops, couldn't we?

Reply:
ok

Reply:
Fixed